components/director/pkg/kubernetes/client.go

```diff
@@ -0,0 +1,57 @@
+package kubernetes
+
+import (
+	"path/filepath"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+type Config struct {
+	PollInterval time.Duration `envconfig:"default=2s"`
```

Comment:
Yes, maybe it's a good idea, the only drawback here would be that if we make those settings come from the env, we would have to describe those environment variables in the `tenant-fetcher-job.yaml` and in the global `values.yaml` file for the whole chart. In the `values.yaml` we describe multiple tenant fetcher jobs and in every single one of them, we would have to also add those 3 properties: K8s timeout, poll interval, and poll timeout. So it can get a bit messy. What do you think?

Reply:
I still think it's a good idea to parameterize this as we might need to tweak those values on demand during issues. Definitely it would be necessary to add the env values to the tenant fetcher job. Regarding the values yaml - if you don't want to duplicate the config for every job, you can introduce a seperate `global.tenaantFetcher.clusterConfig` or something like that which is used by all tenant fetchers. Or you could just define the config in all jobs as you said. Not really sure which I like more.